File: suites/05_ml_primitives/comparisons/tensor_matmul.go
package main
import ("fmt"; "os"; "strconv")
func matmulScore(size int64) int64 {
	var total int64
	for row := int64(0); row < size; row++ {
		for col := int64(0); col < size; col++ {
			var cell int64
			for inner := int64(0); inner < size; inner++ {
				cell += ((row + inner) % 31) * ((inner + col) % 29)
			}
			total += cell
		}
	}
	return total
}
func main() {
	size := int64(96)
	if len(os.Args) > 1 { if v, err := strconv.ParseInt(os.Args[1], 10, 64); err == nil { size = v } }
	fmt.Println(matmulScore(size))
}

File: suites/05_ml_primitives/comparisons/tensor_matmul.rs
const DEFAULT_SIZE: i64 = 96;
fn matmul_score(size: i64) -> i64 {
    let mut total = 0;
    for row in 0..size { for col in 0..size { let mut cell = 0;
        for inner in 0..size { cell += ((row + inner) % 31) * ((inner + col) % 29); }
        total += cell;
    }}
    total
}
fn main() {
    let size = std::env::args().nth(1).and_then(|v| v.parse::<i64>().ok()).unwrap_or(DEFAULT_SIZE);
    println!("{}", matmul_score(size));
}
